jupyterhub/services/auth.py

```diff
@@ -371,9 +371,13 @@ def _api_request(self, method, url, **kwargs):
             )
             app_log.warning(r.text)
             msg = "Failed to check authorization"
-            # pass on error_description from oauth failure
+            # pass on error from oauth failure
             try:
-                description = r.json().get("error_description")
+                description = (
+                    r.json().get("error")
+                    if r.json().get("error")
+                    else r.json().get("error_description")
+                )
```

Comment:
I think error_description should be the more specific explanation and preferred, if defined, in which case the priority should be reversed:

```suggestion
                response = r.json()
                # prefer more specific 'error_description', fallback to 'error'
                description = response.get("error_description", response.get("error", "Unknown error"))
```

Reply:
Ah great point, I'll push an update based on your suggestion. Thanks!